compiler/rustc_ast_passes/src/errors.rs

```diff
@@ -221,6 +221,13 @@ pub enum ExternBlockSuggestion {
 pub struct InvalidSafetyOnExtern {
     #[primary_span]
     pub item_span: Span,
+}
+
+#[derive(Diagnostic)]
+#[diag(ast_passes_extern_invalid_safety_with_suggestion)]
+pub struct InvalidSafetyOnExternWithSuggestion {
```

Comment:
there is no reason to have two of these diagnostics if all that differs is a suggestion

you can make the suggestion optional by using an `Option<Span>`

Reply:
Makes total sense, I was wondering about that but I wasn't sure how to supress the suggestion. Si diagnostics subsystem automatically handle options by not doing anything is None and using the contents when Some(..) I'd assume. Don't have the computer to check but will fix this as soon as I get it. Thanks.